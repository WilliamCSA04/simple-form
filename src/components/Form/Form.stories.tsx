import * as React from 'react';
import Form from '.';
import { GlobalStyles } from '../../styles';

export default {
  title: 'Form',
  decorators: [
    (storyFn: Function) => (
      <>
        <GlobalStyles />
        {storyFn()}
      </>
    ),
  ],
};

export const Default = () => (
  <Form onSubmit={(e) => e.preventDefault}>
    <input value={'test'} />
    <button>Press me</button>
  </Form>
);
