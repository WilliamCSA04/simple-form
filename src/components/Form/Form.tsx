import React from 'react';
import styled from 'styled-components';

export interface Props {
  onSubmit: (e: React.FormEvent<HTMLFormElement>) => void;
  children?: React.ReactNode;
  className?: string;
  id?: string;
}

const StyledForm = styled.form`
  display: flex;
  flex-direction: column;
  width: 300px;
  border-radius: 3px;
  background-color: #3d3b8e;
  padding: 1em;
  margin: auto;
  & > * {
    margin: 1em;
  }
`;

export default function Form({ children, onSubmit, className, id }: Props) {
  return (
    <StyledForm id={id} className={className} onSubmit={onSubmit}>
      {children}
    </StyledForm>
  );
}
