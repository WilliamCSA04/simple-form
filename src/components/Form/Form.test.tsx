import React from 'react';
import { render } from '@testing-library/react';
import Form from '.';

describe('<Form /> snapshot', () => {
  it('renders the component with default props', () => {
    const container = render(<Form id="Test" onSubmit={() => {}} />);
    expect(container.asFragment()).toMatchSnapshot();
  });
});
