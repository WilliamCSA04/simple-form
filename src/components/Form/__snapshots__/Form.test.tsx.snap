// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> snapshot renders the component with default props 1`] = `
<DocumentFragment>
  <form
    id="Test"
  />
</DocumentFragment>
`;
